### ISSUE TITLE
Implementing Linear Regression using Gradient Descent

### ISSUE THREAD

**Issue by lalith1403** [section-io/engineering-education#1165]
### Brief Summary: 
Describe the what, why, and how of your content idea in 2-5 sentences.
In the previous article, we derived the mathematical equations to find the line of best fit iteratively. In this article, I will code linear regression from scratch using numpy, and the equations derived earlier.
### Key Takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article?
1. An overview of the mathematical equations.
2. Code the linear regression class
3. Define the various methods such as optimize, initialize_weights_and_bias, _computeError, optimize, etc., to perform computations as seen in the mathematical equations. 
4. Test the algorithm on the Boston housing dataset and compare the performance with the sklearn library.
5. Going through this article, one will understand how to convert mathematical equations to Python code using basic libraries such as NumPy. 
### References: 
Please list links to any published content/research that you intend to use to support/guide this article. (If none, please indicate 'N/A'.)
N/A

**Comment by ninjaginja** [comment]
good topic @lalith1403 - approved
